src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultIEnumerableConstructibleConverter.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class DefaultIEnumerableConstructibleConverter : JsonEnumerableConverter
+    {
+        public override IEnumerable CreateFromList(Type enumerableType, Type elementType, IList sourceList)
+        {
+            Array array = CreateArrayFromList(elementType, sourceList);
```

Comment:
For performance, instead of creating a strongly-typed array (that has to copy all of the elements) consider a strongly-typed enumerator adapter instead.  I believe just a small wrapper that uses makegenerictype and does a `yield` on each item (casts it to `<T>` which would be `elementType` here). You'd have to measure the perf of that; for large collections it would save more.

Reply:
I added a wrapper for this in the JsonEnumerableConverter class for now as `JsonPropertyInfo` has a generic parameter for the TRuntimeProperty but not the element type. 
I added a to-do to cache the reflection there or add an element type parameter. I'll follow up on this.